sklearn/linear_model/_sag_fast.pyx.tp

```diff
@@ -618,6 +626,17 @@ cdef int lagged_update{{name_suffix}}(
     int n_iter
 ) nogil:
     """Hard perform the JIT updates for non-zero features of present sample.
+
+    In SAG(A), the average-gradient update is collinear with the drawn sample
+    X_i. Therefore, if the data is sparse, the random sample X_i will change
+    the average gradient only on features j where X_ij != 0. In some cases,
+    the average gradient on feature j might change only after k random samples
+    with no change. In these cases, instead of applying k times the same
+    gradient step on feature j, we apply the gradient step only once, scaled by
+    k. This is called the "Just-in-time (JIT) update", and it is performed in
+    this function. This function also applies the prox operator after the
+    gradient step if necessary.
```

Comment:
Thanks for clarifying the implementation, @TomDLT.

I think this comment gives a good overview and better be read first.
Yet I am unsure if it will be read first. Should it be moved at the head of the file instead?

Reply:
Good point. I moved the explanations to `sag{{name_suffix}}`'s docstring, and added some explanations about wscale.